dex/ws/wslink.go

```diff
@@ -101,20 +123,47 @@ func (c *WSLink) Send(msg *msgjson.Message) error {
 		return err
 	}
 
-	c.sendWG.Add(1)
-	c.outChan <- b
-	go func() {
-		c.writeMtx.Lock()
-		b := <-c.outChan
-		c.conn.SetWriteDeadline(time.Now().Add(writeWait))
-		err := c.conn.WriteMessage(websocket.TextMessage, b)
-		c.writeMtx.Unlock()
-		c.sendWG.Done()
-		if err != nil {
-			c.Disconnect()
-		}
-	}()
+	// NOTE: without the stopped chan or access to the Context we are now racing
+	// after the c.Off check above.
+
+	select {
+	case c.outChan <- &sendData{b, writeErr}:
+	case <-c.stopped:
+		return ErrPeerDisconnected
+	}
```

Comment:
@buck54321 I squashed in a refactor of send and outHandler with the previous SendNow commit... breaking my own rules about squashing during review, but the SendNow thing really doesn't belong in this PR anyway.  We can take this whole commit out into another PR if that would be better for review.